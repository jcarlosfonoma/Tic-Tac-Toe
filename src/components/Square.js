import styled from "styled-components";

const lightBlue = "rgb(135,206,250)";
const lightYellow = "rgb(254,228,143)";

const StyledButton = styled.button`
  float: left;
  font-size: 70px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  line-height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 100px;
  height: 100px;
  margin: 4px;
  border: none;
  border-radius: 12px;
  background-color: rgb(61, 66, 80);
  color: ${(props) => (props.children === "X" ? lightBlue : lightYellow)};
`;

function Square(props) {
  return <StyledButton data-testid="square-btn" onClick={props.onClick}>{props.value}</StyledButton>;
}

export default Square;
