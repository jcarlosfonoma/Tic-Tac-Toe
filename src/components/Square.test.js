
import { cleanup, fireEvent, render } from "@testing-library/react";
import Square from "./Square";

describe("Square component", () => {
  afterEach(cleanup);

  it("Square should take a snapshot", () => {
    const { asFragment } = render(<Square />);
    expect(asFragment(<Square />)).toMatchSnapshot();
  });

  it("Checks if initial value is equal to \"\"", () => {
    const { getByTestId } = render(<Square />);
    expect(getByTestId("square-btn")).toHaveTextContent("");
  });

  it("Checks if onClick event is called", () => {
    const onClick = jest.fn();

    const { getByTestId } = render(<Square onClick={onClick} />);

    fireEvent.click(getByTestId("square-btn"));
    expect(onClick).toHaveBeenCalled();
  });
});