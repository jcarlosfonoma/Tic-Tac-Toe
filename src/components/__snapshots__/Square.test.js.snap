// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component Square should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL fPtRnW"
    data-testid="square-btn"
  />
</DocumentFragment>
`;
